### PR TITLE
Refactor item, volunteer and at_risk_user resources

### PR DESCRIPTION
**Pull request by adumortier**
## Who worked on this

- [x] Alexis
- [ ] Maria
- [ ] Ed
- [ ] Cody

## Type of change

- [ ] Style (Non-breaking Style code)
- [ ] Testing (Non-breaking Testing code)
- [x] Refactor
- [ ] New feature (change which adds functionality)

## Summary of the change.

Link to issue:
- DRY up the `volunteer` resource
- DRY up the `at_risk_user` resource
- DRY up the `items` resource
- added instance and class methods for the `volunteer` and `at_risk_user` model

## Checklist:

- [x] My code follows the Turing style guidelines
- [x] I have performed a self-review of my own code
- [x] I have commented my code if necessary

### CODE REVIEW

File: trackbasket_be/resources/volunteer.py [hunk 1]
```diff
@@ -5,33 +5,27 @@
 class Volunteers(Resource):
 
   def get(self,id):
-    volunteer = db.session.query(Volunteer).filter_by(volunteer_id=id).first()
+    volunteer = Volunteer.find_by_id(id)
```
Comment (adumortier) [inline trackbasket_be/resources/volunteer.py:8]: added the `find_by_id` method to the `volunteer`

Comment (mronauli) [reply trackbasket_be/resources/volunteer.py:8]: i like it!

File: trackbasket_be/resources/at_risk_user.py [hunk 2]
```diff
@@ -11,57 +11,33 @@ def get(self, id):
     at_risk_user = AtRiskUser.find_by_id(id)
     if at_risk_user:
       return at_risk_user.json(), 200
-    else: 
-      return {'data': { 'id': 'at_risk_user', 'attributes': {'error': "User not found"} } }, 400
+    return {'data': { 'id': 'at_risk_user', 'attributes': {'error': 'User not found'} } }, 400
```
[the remark this answers is not in the document]
Comment (mronauli) [reply trackbasket_be/resources/at_risk_user.py:14]: super good eye/attention to detail

File: trackbasket_be/resources/at_risk_user.py [hunk 3]
```diff
@@ -11,57 +11,33 @@ def get(self, id):
...
-      at_risk_user.address = data['address']
-      at_risk_user.city = data['city']
-      at_risk_user.state = data['state']
-      at_risk_user.zipcode = data['zipcode']
-      at_risk_user.phone_number = data['phone_number']
-    else: 
-      return {'data': { 'id': 'at_risk_user', 'attributes': {'error': "User not found"} } }, 400
+    if at_risk_user is None:
+      return {'data': { 'id': 'at_risk_user', 'attributes': {'error': 'User not found'} } }, 400
+    at_risk_user.set_attrs(name = data['name'], address = data['address'],
```
[the remark this answers is not in the document]
Comment (mronauli) [reply trackbasket_be/resources/at_risk_user.py:32]: this is neat, i haven't seen it before! thank you for sharing

**Pull request merged**